addons/stock/models/stock_scrap.py

```diff
@@ -47,7 +47,7 @@ class StockScrap(models.Model):
         domain="[('scrap_location', '=', True)]", check_company=True, readonly=False)
     scrap_qty = fields.Float(
         'Quantity', required=True, states={'done': [('readonly', True)]}, digits='Product Unit of Measure',
-        compute='_compute_scrap_qty', default=0.0, readonly=False, store=True)
+        compute='_compute_scrap_qty', default=1.0, readonly=False, store=True)
```

Comment:
I would rather remove the `default` and improve the `compute` to keep the 0-value when no product

Reply:
I understand for the ``compute`` which set ``scrap_qty`` to 0-value when there is no product when we are editing the form view  but on save the method is not hit because ``scrap_qty`` is not precomputed. Without a default value we get a NULL column INSERT error

Reply:
Damn it